### DISCUSSION TITLE
如何完美解析含有 LateX 的页面

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#1553] [使用技巧] [answered]
>简悦 [1.1.2.5005](http://ksria.com/simpread/welcome/version_1.1.2.5005.html#puread) 开始已经可以完美的解析 LateX 页面。

> 简悦会自动检测当前页面是否含有 LateX 但由于每个页面的情况完全不一样，所以并不代表一定能检测成功。

> 如果你经常访问的页面 **无法通过检测 LateX** 而生成阅读模式的话，可以看看下面这两种方式。

***

> 禁用 [预加载机制](http://ksria.com/simpread/docs/#/预加载机制) ，位置 选项页 → 高级设定 → 词法分析引擎

![image](https://user-images.githubusercontent.com/81074/60067596-d1319e80-973d-11e9-8772-3a7663c56c0d.png)

1. 打开出无法显示 LateX  的页面，如 https://blog.csdn.net/VictoriaW/article/details/70053790

2. （确保）等到 **页面完全加载完毕**

3. 进入阅读模式（ 由于启用了 `延迟加载` ，所以进入阅读模式会稍慢些 ）

4. 就能得到完美的 解析 LateX 页面了

*** 

>  启用 [预加载机制](http://ksria.com/simpread/docs/#/预加载机制) ，位置 选项页 → 高级设定 → 词法分析引擎

0. 确保启用下面几个选项，位置 选项页 → 高级设定 → 词法分析引擎

![image](https://user-images.githubusercontent.com/81074/60067423-025d9f00-973d-11e9-8ea1-a7a4f0322911.png)

1. 将 `无法显示 LateX  的 URL ` 加入到 [延迟加载列表](http://ksria.com/simpread/docs/#/预加载机制?id=延迟加载) 位置 选项页 → 高级设定 → 词法分析引擎

   **PS：如果是 csdn.net 的话，请加入 `blog.csdn.net` 即可。**

   ![image](https://user-images.githubusercontent.com/81074/60067298-ad218d80-973c-11e9-9c7a-52bfce24b524.png)

2. 打开无法显示 LateX  的页面，如 https://blog.csdn.net/VictoriaW/article/details/70053790

3. （确保）等到 **页面完全加载完毕** （或手动点击 **展开阅读全文** ）

4. 进入阅读模式（ 由于启用了 `延迟加载` ，所以进入阅读模式会稍慢些 ）

5. 就能得到完美的 解析 LateX 页面了

***

![image](https://user-images.githubusercontent.com/81074/60067377-de01c280-973c-11e9-88aa-e5878be25ac5.png)

## 注意

如果你安装了某些修改页面内容的插件会导致无法正确解析，如：

- [Live Editor](https://simpread.ksria.cn/plugins/details/y8Mai5IBwN) 可以使用，但不能开启 **自动进入 Live Editor 模式**。

- [去除 CSDN 某些字体无法更改的错误](https://simpread.ksria.cn/plugins/details/SumEaxStWE) 不可以使用。

**Comment by B1INGO** [comment]
2021.12.22日 又无法正确显示文章中提到的博客：https://blog.csdn.net/VictoriaW/article/details/70053790
现在这么显示：
![image](https://user-images.githubusercontent.com/20365172/147040991-efdfe659-403e-40cc-861e-b7e5bf40c283.png)
我的简悦相关配置：
![image](https://user-images.githubusercontent.com/20365172/147041092-ff810911-ffbb-493e-826e-f6e2d9e6e4b2.png)
插件安装情况：
![image](https://user-images.githubusercontent.com/20365172/147041138-e307e43f-6e17-4831-8ffa-43a91eeb0a08.png)

**Reply by Kenshin** [reply]
我这边没问题，请确认是否可以稳定复现？

或者试试**隐身模式下仅使用简悦的方式** 看看能否解决？

![image](https://user-images.githubusercontent.com/81074/147047568-17f59fd2-c62d-4cbe-bb9f-d01f325eb65b.png)

**Reply by Kenshin** [reply]
@B1INGO 

嗨，我补充了一部分无法生效的说明，估计是这个问题造成的。

![image](https://user-images.githubusercontent.com/81074/147743533-a75bcdf1-ed9d-4e1c-9b2c-ca3e06839530.png)

**Comment by An-Jhon** [comment]
开启和关闭是一样的效果

**Reply by Kenshin** [reply]
请提供 URL 否则我没法判断。

**Reply by An-Jhon** [reply]
URL就是上文提到的博客地址，
<img width="1114" alt="image" src="https://github.com/Kenshin/simpread/assets/44490800/ee1f8e45-f803-40cc-9a1f-2cd2919b73da">
另外其他地址我也尝试了，同样的问题：

**Reply by An-Jhon** [reply]
这个url中有同样的问题：https://www.cnblogs.com/pinard/p/10979808.html
<img width="1003" alt="image" src="https://github.com/Kenshin/simpread/assets/44490800/bb35aff6-d2c1-4a7c-8bcd-f713f52accc8">

**Reply by An-Jhon** [reply]
还有就是在我的个人博客（基于NotionNext）好像获取不到公式和代码：https://www.anjhon.top/article/077c14d7-9ae0-4e05-832d-acb33df9fca8
![image](https://github.com/Kenshin/simpread/assets/44490800/39159d57-189d-47e3-b831-efeaefd9243a)

**Reply by Kenshin** [reply]
我看了 cnblog 使用的仍是 MathJax 方案，所以没有任何问题。

所以，请检查下自己的设置。

> 测试网页 https://www.cnblogs.com/pinard/p/10979808.html

<img width="1252" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/3288495b-3824-4d74-9d05-2546d850af3f">

> 还有就是在我的个人博客（基于NotionNext）好像获取不到公式和代码：https://www.anjhon.top/article/077c14d7-9ae0-4e05-832d-acb33df9fca8

这个页面使用了 `katax` 方案，简悦暂时不支持此方式，目前仅支持 `MathJax` 方案。

**Comment by findyy99** [comment]
开发者，您好，我这边为啥阅读模式没法正常渲染latex公式，网站是mathjax3.2.0，如图：
<img width="500" alt="image" src="https://github.com/Kenshin/simpread/assets/26852735/19d61f8a-5c8d-4748-bd17-522f1a877a59">
- 网址链接是这个https://www.learndatasci.com/glossary/gini-impurity/
- 词法分析设置如图：
<img width="500" alt="image" src="https://github.com/Kenshin/simpread/assets/26852735/ab2336c6-6c9f-4ad7-b364-dc2a08eef96b">

**Reply by Kenshin** [reply, marked as answer]
因为这个站没有适配，没有正确获取正文导致的，你可以按下图所示的方式手动方案

<img width="752" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/ab263c0a-77c0-460b-a556-e95bfc6f951f">

生成阅读模式后就会出现公式了

<img width="1267" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/455952aa-a719-4a92-bbb1-15114a01c506">

关于这个功能的更多说明 https://github.com/Kenshin/simpread/discussions/3340

***

PS：你没有把这个 URL 加入到延迟加载列表（截图没有体现），但由于可以使用手动方式来生成阅读模式，所以也可以不用加入延迟加载列表。

**Reply by findyy99** [reply]
感谢🙏

**Comment by Ufitok** [comment]
作者你好，我也遇到了对latex解析完成后，导出markdown、pdf发现公式丢失的情况。以pdf为例，下面我展示了浏览器简悦界面和pdf界面，发现到处pdf中公式丢失。请问这个问题能够解决吗，谢谢？
<img width="789" alt="1720437762973" src="https://github.com/Kenshin/simpread/assets/93562154/d7abc157-db3a-439e-9e42-4ecb536e5905">
<img width="702" alt="1720437753344" src="https://github.com/Kenshin/simpread/assets/93562154/8d75b3e9-e935-4ebb-a1e3-fc186e959991">

**Reply by Kenshin** [reply]
okay 收到~

需要提供出问题的 URL

**Reply by Ufitok** [reply]
https://www.sciencedirect.com/science/article/pii/S0377221721009541?via=ihub#sec0003
这是我上面示例的文献网址，您看一下。
另外，还有一个网址，https://onlinelibrary.wiley.com/doi/10.1002/nav.22173。markdown格式是下图这样的。
总的来说，简悦界面一般来说还是正常的，导出为markdown和pdf时，公式会丢失或者不能够对应添加$公式符号。
<img width="969" alt="1720439751343" src="https://github.com/Kenshin/simpread/assets/93562154/c67cfe89-05d7-4de1-94b7-c4978298f016">

**Reply by Kenshin** [reply]
okay

**Reply by qingmu-z** [reply]
https://blog.csdn.net/qq_43653581/article/details/105171383 你好，这个页面公式混乱
![image](https://github.com/user-attachments/assets/1d4f6234-4e62-411a-a54c-f0198c4e5eb5)
![image](https://github.com/user-attachments/assets/d2db80da-471d-4352-8217-e1499f56cae1)
前面的csdn示例公式是没有问题的

**Reply by Kenshin** [reply]
@qingmu-z 

嗨，谢谢提交。

这是预期情况 😂 因为这个页面使用了 Katex 方案，简悦仅支持 MathJax 1.x 的部分版本。

我先记录下，解决后后在这里通知你~

**Comment by FlendyWhite** [comment]
求助，稍后读Latex解析有问题。
原网页：https://www.cnblogs.com/leimu/p/13949945.html
公式内容：
<img width="309" height="137" alt="原网页" src="https://github.com/user-attachments/assets/b74bef5b-b713-41da-afdf-e261841c2082" />
阅读模式下：
<img width="418" height="316" alt="阅读模式" src="https://github.com/user-attachments/assets/5ff50c77-0dab-41cb-8c7a-2fb5165ac303" />
稍后读里：
<img width="751" height="214" alt="稍后读里" src="https://github.com/user-attachments/assets/6baba97b-2e5d-4704-abe6-86b4888a3948" />
简悦设置：
<img width="510" height="290" alt="简悦设置" src="https://github.com/user-attachments/assets/75728f74-e9ee-4a09-9041-4dc8273fc67f" />
自动同步已配置：
<img width="568" height="416" alt="自动同步已配置" src="https://github.com/user-attachments/assets/842dba11-55af-4b29-aa86-30dd8e1944f1" />
稍后读设置：
<img width="431" height="345" alt="稍后读设置" src="https://github.com/user-attachments/assets/0d0dfd62-c6dd-4388-b3ab-97249448ec79" />

**Reply by Kenshin** [reply]
在稍后读完美支持 Latex 需要使用本地快照系统，也就是需要使用同步助手，教程（内含一站式设置）https://www.yuque.com/kenshin/simpread/wkswh7
